homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -128,21 +144,26 @@ async def async_setup_scanner(
     if request_rssi:
         _LOGGER.debug("Detecting RSSI for devices")
 
-    async def perform_bluetooth_update():
+    async def perform_bluetooth_update() -> None:
         """Discover Bluetooth devices and update status."""
         _LOGGER.debug("Performing Bluetooth devices discovery and update")
         tasks = []
 
         try:
             if track_new:
                 devices = await hass.async_add_executor_job(discover_devices, device_id)
-                for mac, device_name in devices:
-                    if mac not in devices_to_track and mac not in devices_to_not_track:
-                        devices_to_track.add(mac)
+                for device in devices:
+                    if (
+                        device[1] not in devices_to_track
+                        and device[1] not in devices_to_not_track
+                    ):
+                        devices_to_track.add(device[1])
 
             for mac in devices_to_track:
-                device_name = await hass.async_add_executor_job(lookup_name, mac)
-                if device_name is None:
+                device_lookuped_name: str | None = await hass.async_add_executor_job(
```

Comment:
This name looks a bit cumbersome. What was wrong with the old one?

Reply:
before b494c57 `device_name` was already defined some lines above as `str` and those here it comes to typing mismatch, since it could be `Optional[str]` but variabel is `str` ... but yes, now it can be re-renamed

Reply:
What pylint errors did you have there? I think old code was more idiomatic.

Reply:
with
```python
        try:
            if track_new:
                devices = await hass.async_add_executor_job(discover_devices, device_id)
                for mac, device_name in devices:
                    if mac not in devices_to_track and mac not in devices_to_not_track:
                        devices_to_track.add(mac)

            for mac in devices_to_track:
                device_name = await hass.async_add_executor_job(lookup_name, mac)
                if device_name is None:
                    # Could not lookup device name
                    continue
```
get `error: Argument 1 to "async_add_executor_job" of "HomeAssistant" has incompatible type "Callable[[str], Optional[str]]"; expected "Callable[..., str]"  [arg-type]`, becasue `device_name` from for loop is of typ `str`, but `lookup_name()` has returntype `Optional[str]`.
strictly typing `device_name: str | None = await...` leads to an "already defined before" error (_`error: Name 'device_name' already defined on line 156  [no-redef]`_)

Reply:
So just this should be enough right? "Friendly name" comes from [PyBluez docs](https://pybluez.readthedocs.io/en/latest/api/lookup_name.html).
```python
friendly_name = await hass.async_add_executor_job(lookup_name, mac)
```

Reply:
using any other than `device_name` to store result of `lookup_name` will solve it, but on the other hand lead to an pylint error about unused variable, since `device_name` is never used within the loop ... it is a little bit like catching the rabbit :smile:

Reply:
I see that it's used on lines 166 and  177.

Reply:
https://github.com/home-assistant/core/blob/b494c570a0c6b84d47998f5f5fad50634ebe5485/homeassistant/components/bluetooth_tracker/device_tracker.py#L147-L184

if line 163 is changed to use something other than `device_name`- let's take your proposal
```python
friendly_name = await hass.async_add_executor_job(lookup_name, mac)
```
`device_name` has also to be replaced in lines 166 and 177 and those `device_name` defined in loop is never used

Reply:
maybe simple ignore the pylint "unused variable" error :thinking:

Reply:
> than device_name has also to be replaced in lines 166 and 177 

Yes, that's what I mean.

> and those device_name defined in loop is never used

This one I don't get.

Here is my version:
```python
            if track_new:
                devices = await hass.async_add_executor_job(discover_devices, device_id)
                for mac, _device_name in devices:
                    if mac not in devices_to_track and mac not in devices_to_not_track:
                        devices_to_track.add(mac)

            for mac in devices_to_track:
                friendly_name = await hass.async_add_executor_job(lookup_name, mac)
                if friendly_name is None:
                    # Could not lookup device name
                    continue

                rssi = None
                if request_rssi:
                    client = BluetoothRSSI(mac)
                    rssi = await hass.async_add_executor_job(client.request_rssi)
                    client.close()

                tasks.append(see_device(hass, async_see, mac, friendly_name, rssi))
```

Btw, just noticed that your version contains an error. It should be `device[0]`.

Reply:
a simple `_` prefix marks a variable as unused to pylint? good to know :slightly_smiling_face: 
I like your version :+1:

Reply:
Yes, it's a convention that names of unused variables start with `_`. But for class attributes underscore has different meaning, it means that attribute is private. This is a bit confusing.

Reply:
great ... again, new things learned :heart:

Reply:
Btw, here is pro tip for you about mypy.

You can insert `reveal_type(variable)` calls inside the code and mypy will print current type, no extra imports required. Don't forget to clean it up because this function only exists when running mypy.